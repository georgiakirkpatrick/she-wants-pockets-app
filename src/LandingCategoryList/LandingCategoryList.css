h2 {
    display: block;
    font-size: 1.5em;
    margin-block-start: 0.5em;
    margin-block-end: 0.5em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    font-weight: bold;
}

.LandingCategoryList__list {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
    margin: 1em 0 0;
    margin-block-start: 0;
    margin-block-end: 0;
    padding-inline-start: 0;
}

.LandingCategoryList__section {
    padding: 0 15px;
    padding: 20px 10px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    font: arial;
}