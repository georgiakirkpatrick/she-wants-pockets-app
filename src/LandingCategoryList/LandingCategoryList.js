import React from 'react';
import Category from '../Category/Category';
import ReadMoreLink from '../ReadMoreLink/ReadMoreLink';
import './LandingCategoryList.css';

function LandingCategoryList(props) {
    const generateCategoryList = props.categories.map(category => (
        <Category 
            key={category.id}
            id={category.id}
            pathToImage={category.picPath}
            imgAlt={category.imgAlt}
            categoryName={category.name}
            categoryLink={category.link}
        />
    ))

    return (
        <section className='LandingcategoryList__section'>
            <header>
                <h3>
                    Use filters to narrow your search
                </h3>
            </header>
            <p>
                Search by design feature, size, brand, fiber, and other details to find 
                the fashion you're looking for.
            </p>
            <ul className='LandingCategoryList__list'>
                {generateCategoryList}
            </ul>
            <ReadMoreLink 
                text='More categories' 
                link='/all-categories'
            />
        </section>
    )
}

LandingCategoryList.defaultProps = {
    categories: []
}

export default LandingCategoryList;