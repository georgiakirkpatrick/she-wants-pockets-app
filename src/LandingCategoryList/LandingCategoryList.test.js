import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import LandingCategoryList from './LandingCategoryList'

const categories = [
    {
        id: 3694424,
        pathToImage: 'https://res.cloudinary.com/everlane/image/upload/c_scale/c_fill,dpr_1.0,f_auto,q_auto,w_auto:100:1200/v1/i/1bc014ee_3f92.jpg',
        altText: `A 5'10" model is wearing a size 2 The Relaxed Air Shirt in Blue / White Stripe`,
        category: 'Tops'
    }
]

describe ('LandingCategoryList', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Router><LandingCategoryList /></Router>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Router><LandingCategoryList /></Router>)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})