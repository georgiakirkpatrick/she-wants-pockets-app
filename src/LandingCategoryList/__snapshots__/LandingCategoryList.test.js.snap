// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingCategoryList renders the UI as expected 1`] = `
<section
  className="LandingcategoryList__section"
>
  <header>
    <h3>
      Use filters to narrow your search
    </h3>
  </header>
  <p>
    Search by design feature, size, brand, fiber, and other details to find the fashion you're looking for.
  </p>
  <ul
    className="LandingCategoryList__list"
  >
    <li>
      <a
        className="main-link"
        href="www.google.com"
      >
        <div
          className="Category"
        >
          <img
            alt="alternate text"
            src="www.google.com"
          />
          <p>
            tops
          </p>
        </div>
      </a>
    </li>
  </ul>
  <a
    href="/all-categories"
  >
    <p
      className="ReadMoreLink"
    >
      <span>
        More categories
         
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 "
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </p>
  </a>
</section>
`;
